client/galaxy/scripts/mvc/rules/rule-definitions.js

```diff
@@ -713,7 +713,7 @@ const MAPPING_TARGETS = {
         help: _l(
             "If this is set, all rows with the same collection name will be joined into a collection and it is possible to create multiple collections at once."
         ),
-        modes: ["raw", "ftp"], // TODO: allow this in datasets mode & tool builder modes
+        modes: ["raw", "ftp", "datasets", "library_datasets"],
```

Comment:
you deleted this whole line in https://github.com/galaxyproject/galaxy/pull/5969, is this the proper merge?

Reply:
Yes... thanks for looking so close though! Think of modes like a filter - in #5969 I made it so all the modes of the rule builder that produce collections could production multiple collections so I didn't need that any more. This PR adds the ability to apply to existing collections using a tool that only has one output - so I needed to re-list all the valid modes and exclude that one new mode ("collection_contents").

<img width="825" alt="screen shot 2018-04-24 at 4 14 41 pm" src="https://user-images.githubusercontent.com/216771/39211891-0358ffb0-47db-11e8-9617-02fd2a57b194.png">